### PR TITLE
ExcelsiorJetPlugin simplification

### PR DESCRIPTION
**Pull request by Crazyjavahacking**
Done during investigation of the plugin.

### CODE REVIEW

File: src/main/groovy/com/excelsiorjet/gradle/plugin/ExcelsiorJetPlugin.groovy [hunk 1]
```diff
@@ -58,34 +58,36 @@ class ExcelsiorJetPlugin implements Plugin<Project> {
 
         extension = target.getExtensions().create(ExcelsiorJetExtension.EXTENSION_NAME, ExcelsiorJetExtension)
 
-        def archiveTaskName
+        Task archiveTask
```
Comment (Crazyjavahacking) [inline src/main/groovy/com/excelsiorjet/gradle/plugin/ExcelsiorJetPlugin.groovy:61]: It's more efficient. We don't really need archiveTaskName, it is only used for getting the real task.

File: src/main/groovy/com/excelsiorjet/gradle/plugin/ExcelsiorJetPlugin.groovy [hunk 2]
```diff
@@ -95,19 +97,19 @@ class ExcelsiorJetPlugin implements Plugin<Project> {
                 return project.version.toString()
             }
         }
-        extension.conventionMapping.artifactName = { getArchiveName(project.tasks.getByPath(taskPath(project, "jar"))) }
+        extension.conventionMapping.artifactName = { getArchiveName(archiveTask) }
```
Comment (Crazyjavahacking) [inline src/main/groovy/com/excelsiorjet/gradle/plugin/ExcelsiorJetPlugin.groovy:100]: Once we have the `archiveTask`, we can use it directly.

File: src/main/groovy/com/excelsiorjet/gradle/plugin/ExcelsiorJetPlugin.groovy [hunk 3]
```diff
@@ -95,19 +97,19 @@ class ExcelsiorJetPlugin implements Plugin<Project> {
...
         if (!isWar) {
             extension.conventionMapping.mainJar = {
-                new File(project.tasks.getByPath(taskPath(project, "jar")).archivePath as String)
```
Comment (Crazyjavahacking) [inline src/main/groovy/com/excelsiorjet/gradle/plugin/ExcelsiorJetPlugin.groovy:101]: This is not necessary. Plugin will always be applied to a specific project and such project is expected to have `jar` task.

Also `archivePath` already returns a file, so it is unnecessary to convert such file into a String, just to be converted to the File again.

**Comment by pjBooms** [comment]
Thanx for PR, it seems reasonable

**Pull request merged**